hbase-server/src/main/java/org/apache/hadoop/hbase/master/HMaster.java

```diff
@@ -384,6 +387,9 @@ public class HMaster extends HBaseServerBase<MasterRpcServices> implements Maste
   private DirScanPool logCleanerPool;
   private LogCleaner logCleaner;
   private HFileCleaner hfileCleaner;
+  private HFileCleaner[] customHFileCleaners;
+  private Path[] customHFilePaths;
+  private DirScanPool customHFileCleanerPool;
```

Comment:
Thanks. All the custom paths cleaners use a shared DirScanPool, while the origin archive cleaner use an exclusive DirScanPool.  I think currently it is not necessary for each custom path use its unique DirScanPool, since all the customs paths are subdirectories under the hfile archive path.